jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -313,13 +315,27 @@ def _update_mark_scale(self, *args):
         else:
             self.background.scale_factor = self.slice_spectral_value/self.reference_spectral_value
 
+    def _on_gloabl_display_unit_changed(self, msg={}):
+
+        if msg.axis == 'spectral_y':
+            self.spectrum_y_units = str(msg.unit)
+
+        # a 'flux' and 'sb' message should be recieved back to back from
+        # the unit conversion plugin, so don't need to sync them immediatley
+        # within each message recieved
+        if msg.axis == 'flux':
+            self.flux_unit = str(msg.unit)
+        if msg.axis == 'sb':
+            self.sb_unit = str(msg.unit)
+
     @observe('function_selected')
-    def _update_results_units(self, msg={}):
-        self.spectrum_y_units = str(self.app._get_display_unit('spectral_y'))
+    def _update_units_on_function_selection(self, *args):
+
         if self.function_selected.lower() == 'sum':
-            self.results_units = str(self.app._get_display_unit('flux'))
+            self.results_units = self.flux_unit
         else:
-            self.results_units = str(self.app._get_display_unit('sb'))
+            self.results_units = self.sb_unit
```

Comment:
does the old logic not work or are we just trying to avoid calls to `_get_display_unit`?  This seems to be quite a bit more logic that could break

Reply:
the change i made here was because every time a new 'function' was selected, all three display units (sb, flux, and spectral_y) were being obtained again when they shouldn't have changed. additionally, since unit conversion is not broadcasting three global display unit changes when there's a new flux selection, this would be called an additional three times. the way i have it working now is that there are traitlets for spectral_y, flux, and sb that are updated only when there's a unit change, and then upon function change the display unit for the plugin is chosen from these traitlets based on the selection (e.g 'sum' will grab the 'flux' unit)

Reply:
ok, I do see the benefits, so I'm ok with this for now and we can always try to simplify it as the rest of the pieces fall into place.